### DISCUSSION TITLE
Recommended prepare hooks pattern for multiple transactions

### DISCUSSION THREAD

**Discussion by jmrossy** [wevm/wagmi#1564] [Q&A]
Hi, are there recommended patterns for using the prepare + send hooks when more than one transaction is needed from a component, such as a token approve + transfer? 

There are a few ways to one could get this working:
1. Separate the transactions into different components, have the parent switch between then on confirmation of first
2. Use two prepare and two send hooks together, with some state to 'enable' the second, similar to this [example here](https://github.com/wagmi-dev/wagmi/discussions/1529).

These work but feel a bit hacky. Is there another option I'm missing? At the moment, I'm inclined to avoid the hooks entirely and use the `@wagmi/core` methods directly for more fine grained control.

**Comment by fachrudahri** [comment]
I've solved this problem.
what has to be done is when the approve has been a success and the block on approve has been updated, run (watch new block with useEffect) the refetch function on usePrepareContractWrite transfer

**Comment by Soniachniy** [comment]
I've faced the issue when the second transaction didn't pop up Metamask and flows became broken because user don't know about pending transaction in MM. Any variants?

**Reply by jmrossy** [reply]
Yes, I see that same bug. There's been more discussion here but no resolution yet: https://github.com/wagmi-dev/wagmi/discussions/2928

**Reply by avious00** [reply]
@Solniechniy seen the same, yes

**Reply by Soniachniy** [reply]
@jmrossy @avious00 Guys, my problem was resolved by switching to web3modal to v3
